src/utilities/extendSchema.js

```diff
@@ -212,9 +181,6 @@ export function extendSchema(
   const extendTypeCache = Object.create(null);
 
   // Get the root Query, Mutation, and Subscription object types.
-  // Note: While this could make early assertions to get the correctly
-  // typed values below, that would throw immediately while type system
-  // validation with validateSchema() will produce more actionable results.
```

Comment:
This comment is not relevant anymore since they're no `any` cast here.
Moreover, the result of extending object type will always be an object type.